plugins/nextcloud/nextcloud_

```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
+
+set -e
+
+: << =cut
+
+=head1 NAME
+
+nextcloud_ - Monitor usage of nextcloud instances
+
+=head1 APPLICABLE SYSTEMS
+
+Nexcloud instances
+
+=head1 CONFIGURATION
+
+Requires installed curl and jq, a command-line json processor.
+
+This is a wildcard plugin. To monitor a nextcloud instance, link
+nextcloud_<nextcloud-domain> to this file. You can even append a port 
+(:8443) to the file if needed. For example,
+
+  ln -s /usr/share/munin/plugins/nextcloud_ \
+        /etc/munin/plugins/nextcloud_cloud.domain.tld
+
+Set username and password in your munin-node configuration
+
+[nextcloud_cloud.domain.tld]
+env.username <nexcloud_user>
+env.password <nextcloud_password>
+
+It's advised to set an app password (for this plugin) in your nextcloud 
+instance and not to use the "real" password of your nextcloud user.
+
+=head1 AUTHOR
+
+Copyright (C) 2020 Sebastian L. (https://momou.ch)
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=manual
+ #%# capabilities=autoconf
+
+=cut
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+API_PATH="/ocs/v2.php/apps/serverinfo/api/v1/info?format=json"
+DOMAIN="${0##*nextcloud_}"
+CLEANDOMAIN="$(clean_fieldname "$DOMAIN")"
+USERNAME="${username:-}"
+PASSWORD="${password:-}"
+
+print_json_data() {
+    [ -z "$1" ] && exit 0
+    for KEY in "${@:2}"; do
```

Comment:
I think, this is the only place where you use a bash-only feature. Maybe assign `$1` to a local variable and add `shift` in order to void this and reduce the requirements to a simple shell.

Reply:
Thanks for changing the shell to `sh`. Now you need to add `shift` here and use the non-indexed `$@` substitution.

Reply:
Nah ... it was not only this. In bash grepping jq outputs was working, in posix shell I had to escape the slashes. Now - I think - it's fine.

Reply:
> In bash grepping jq outputs was working, in posix shell I had to escape the slashes.

That sounds really strange. I am not aware of differences between the shells regarding pipelined data. I have the vague feeling, that your were stumbling upon something else. Maybe you are curious to investigate that? I would enjoy to hear the details ...

Anyway, I am merging your contribution now. Thank you for going through all the details!

Reply:
Yes, it is strange.

I tried a lot of stuff and I really don't know where "sh" is stumbling about where "bash" is not with this jq ... It was more trial'n'error that I found out that escaping the slahses helps ... I'll dig further ...